util/chplenv/printchplenv.py

```diff
@@ -211,6 +212,11 @@ def compute_all_values():
     chpl_llvm.validate_llvm_config()
     chpl_compiler.validate_compiler_settings()
 
+    if ENV_VALS['CHPL_LOCALE_MODEL'] == 'gpu' and \
+       ENV_VALS['CHPL_COMM'] != "none":
+        error("The prototype GPU support does not work when CHPL_COMM is not set to\n\"none\".");
```

Comment:
Are there any issues with having a `chpl_gpu.validate(comm)` function? I feel like this is unnecessarily inconsistent with similar checks.

Reply:
The reason I didn't do that is I wasn't sure if this really belongs under `chpl_locale_model` or `chpl_comm` (the issues really about an interaction between the two). I hadn't thought about `chpl_gpu` but it isn't any more obvious that it belongs there.

But I agree having the code floating there is kind of ugly. I think I'll take your suggestion and stick it in `chpl_gpu` on the basis that even if that doesn't seem like the "perfect" place for it, it still seems less ugly than having it inlined there.

Reply:
Yeah good point about the place in which that validation belongs. But, yeah, I agree that it is less ugly.